marlowe-chain-sync/chain-indexer/Language/Marlowe/Runtime/ChainIndexer/Store.hs

```diff
@@ -49,8 +49,9 @@ data ChainStoreDependencies r = ChainStoreDependencies
   }
 
 -- | Create a ChainStore component.
-chainStore :: Component IO (ChainStoreDependencies r) ()
-chainStore = component_ \ChainStoreDependencies{..} -> do
+chainStore :: Component IO (ChainStoreDependencies r) (STM ())
+chainStore = component \ChainStoreDependencies{..} -> do
+  genesisReadyVar <- newEmptyTMVar
```

Comment:
What was the requirement that necessitated adding this?

Reply:
There was a `chainseekd` query that was failing because the `chain.block` didn't have any rows yet. Basically, this is a healthcheck for `marlowe-chain-indexer`.

Reply:
@bwbush I found a better way to handle this in the integration test driver https://github.com/input-output-hk/marlowe-cardano/pull/407/commits/d9e10f0bbfefe400ef2cbacee38dd489c6c725dc